service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/RawTwinQuery.java

```diff
@@ -137,9 +218,24 @@ public synchronized String next(Query query) throws IOException, IotHubException
         }
         else
         {
-            //Codes_SRS_RAW_QUERY_25_017: [ If the next element from the query response is an object other than String, then this method shall throw IOException ]
             throw new IOException("Received a response that could not be parsed");
         }
+    }
+
+    private String getAuthenticationToken()
+    {
+        // Three different constructor types for this class, and each type provides either a TokenCredential implementation,
+        // an AzureSasCredential instance, or just the connection string. The sas token can be retrieved from the non-null
+        // one of the three options.
+        if (this.credential != null)
+        {
+            return this.credential.getToken(new TokenRequestContext()).block().getToken();
```

Comment:
Check with service team add add the necessary prefix and suffix - https://github.com/Azure/azure-sdk-for-net/blob/8970711b5c82c9fac3f7135529ab57de5a294f08/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L218